### PR TITLE
bump: support `torch>=2.0`

### LINKED ISSUE #2791: failing tests for PT 1.1x and python 3.10+
### To Reproduce

https://github.com/Lightning-AI/torchmetrics/actions/runs/11427457506/job/31791630547?pr=2790

### Environment

- TorchMetrics version (if build from source, add commit SHA): ???
- Python & PyTorch Version (e.g., 1.0): ???
- Any other relevant information such as OS (e.g., Linux): ???

### Additional context

we really shall bump req. to PT 2.0+ see #2671

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

remove active testing and raising unsupported warnings for older `torch` and related `torchvision` & `torchaudio`
Note that torch 2.0 was released for more than a year ago (Mar 2023) and seems there have been not additional releases bug-fix releases for the 1.X series since Dec 2022

close #2791

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [x] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2671.org.readthedocs.build/en/2671/

### CODE REVIEW

File: requirements/_tests.txt [hunk 1]
```diff
@@ -10,9 +10,9 @@ pytest-timeout ==2.3.1
 pytest-xdist ==3.6.1
 phmdoctest ==1.4.0
 
-psutil <6.1.0
-pyGithub ==2.4.0
-fire <=0.6.0
+psutil ==6.1.*
```
Comment (SkafteNicki) [inline requirements/_tests.txt:13]:
```suggestion
psutil ==6.*
```
v6.0.0 is the newest version

**Review by SkafteNicki — APPROVED** [review]

**Comment by Borda** [comment]
last failing, @SkafteNicki ?
```
    def test_on_real_audio():
        """Test that metric works on real audio signal."""
        rate, ref = wavfile.read(_SAMPLE_AUDIO_SPEECH)
        rate, deg = wavfile.read(_SAMPLE_AUDIO_SPEECH_BAB_DB)
>       assert torch.allclose(
            short_time_objective_intelligibility(torch.from_numpy(deg), torch.from_numpy(ref), rate).float(),
            torch.tensor(0.6739177),
            rtol=0.0001,
            atol=1e-4,
        )
E       assert False
E        +  where False = <built-in method allclose of type object at 0x7f1ccc8cd880>(tensor(0.6720), tensor(0.6739), rtol=0.0001, atol=0.0001)
E        +    where <built-in method allclose of type object at 0x7f1ccc8cd880> = torch.allclose
E        +    and   tensor(0.6720) = <built-in method float of Tensor object at 0x7f1b157a3830>()
E        +      where <built-in method float of Tensor object at 0x7f1b157a3830> = tensor(0.6720, dtype=torch.float64).float
E        +        where tensor(0.6720, dtype=torch.float64) = short_time_objective_intelligibility(tensor([ 863,  991, 1338,  ...,  -83, -476, -294], dtype=torch.int16), tensor([  0,   0,   0,  ..., -12, -11,  -9], dtype=torch.int16), 16000)
E        +          where tensor([ 863,  991, 1338,  ...,  -83, -476, -294], dtype=torch.int16) = <built-in method from_numpy of type object at 0x7f1ccc8cd880>(array([ 863,  991, 1338, ...,  -83, -476, -294], dtype=int16))
E        +            where <built-in method from_numpy of type object at 0x7f1ccc8cd880> = torch.from_numpy
E        +          and   tensor([  0,   0,   0,  ..., -12, -11,  -9], dtype=torch.int16) = <built-in method from_numpy of type object at 0x7f1ccc8cd880>(array([  0,   0,   0, ..., -12, -11,  -9], dtype=int16))
E        +            where <built-in method from_numpy of type object at 0x7f1ccc8cd880> = torch.from_numpy
E        +    and   tensor(0.6739) = <built-in method tensor of type object at 0x7f1ccc8cd880>(0.6739177)
E        +      where <built-in method tensor of type object at 0x7f1ccc8cd880> = torch.tensor

unittests/audio/test_stoi.py:127: AssertionError
```

**Comment by SkafteNicki** [comment]
Needed to reduce an array size in https://github.com/Lightning-AI/torchmetrics/pull/2671/commits/397ecd18d9c5b2284253c9947c0fdf5bbc1ab250 because I found that `torch.unique` behaves weird for newest version of Pytorch on Windows (reported to torch team in issue https://github.com/pytorch/pytorch/issues/135019)

File: tests/unittests/detection/test_intersection.py [hunk 2]
```diff
@@ -24,16 +24,10 @@
...
-    from torchvision.ops import box_iou as tv_iou
-    from torchvision.ops import complete_box_iou as tv_ciou
-    from torchvision.ops import distance_box_iou as tv_diou
-    from torchvision.ops import generalized_box_iou as tv_giou
-else:
-    tv_iou, tv_ciou, tv_diou, tv_giou = ..., ..., ..., ...
+from torchvision.ops import box_iou as tv_iou
+from torchvision.ops import complete_box_iou as tv_ciou
+from torchvision.ops import distance_box_iou as tv_diou
+from torchvision.ops import generalized_box_iou as tv_giou
```
Comment (Borda) [inline tests/unittests/detection/test_intersection.py:30]:
```suggestion
from torchmetrics.utilities.imports import _TORCHVISION_AVAILABLE

if _TORCHVISION_AVAILABLE:
    from torchvision.ops import box_iou as tv_iou
    from torchvision.ops import complete_box_iou as tv_ciou
    from torchvision.ops import distance_box_iou as tv_diou
    from torchvision.ops import generalized_box_iou as tv_giou
else:
    tv_iou, tv_ciou, tv_diou, tv_giou = ..., ..., ..., ...
```

File: src/torchmetrics/detection/__init__.py [hunk 3]
```diff
@@ -11,23 +11,19 @@
...
-    __all__ += ["CompleteIntersectionOverUnion", "DistanceIntersectionOverUnion"]
+__all__ = [
+    "ModifiedPanopticQuality",
+    "PanopticQuality",
+    "MeanAveragePrecision",
+    "GeneralizedIntersectionOverUnion",
+    "IntersectionOverUnion",
+    "CompleteIntersectionOverUnion",
+    "DistanceIntersectionOverUnion",
+]
```
Comment (Borda) [inline src/torchmetrics/detection/__init__.py:29]:
```suggestion
from torchmetrics.utilities.imports import _TORCHVISION_AVAILABLE

__all__ = ["ModifiedPanopticQuality", "PanopticQuality"]

if _TORCHVISION_GREATER_EQUAL_0_8:
    from torchmetrics.detection.giou import GeneralizedIntersectionOverUnion
    from torchmetrics.detection.iou import IntersectionOverUnion
    from torchmetrics.detection.mean_ap import MeanAveragePrecision
    from torchmetrics.detection.ciou import CompleteIntersectionOverUnion
    from torchmetrics.detection.diou import DistanceIntersectionOverUnion

    __all__ += ["MeanAveragePrecision", "GeneralizedIntersectionOverUnion", "IntersectionOverUnion", "CompleteIntersectionOverUnion", "DistanceIntersectionOverUnion"]
```

File: src/torchmetrics/detection/__init__.py [hunk 4]
```diff
@@ -11,23 +11,19 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from torchmetrics.detection.ciou import CompleteIntersectionOverUnion
+from torchmetrics.detection.diou import DistanceIntersectionOverUnion
+from torchmetrics.detection.giou import GeneralizedIntersectionOverUnion
+from torchmetrics.detection.iou import IntersectionOverUnion
+from torchmetrics.detection.mean_ap import MeanAveragePrecision
```
Comment (Borda) [inline src/torchmetrics/detection/__init__.py:18]:
```suggestion
```

File: src/torchmetrics/detection/ciou.py [hunk 5]
```diff
@@ -110,11 +110,6 @@ def __init__(
         respect_labels: bool = True,
         **kwargs: Any,
     ) -> None:
```
Comment (Borda) [inline src/torchmetrics/detection/ciou.py:112]:
```suggestion
    ) -> None:
        if not _TORCHVISION_AVAILABLE:
            raise ModuleNotFoundError(
                f"Metric `{self._iou_type.upper()}` requires that `torchvision` to be installed."
                " Please install with `pip install torchvision -U` or `pip install torchmetrics[detection]`."
            )
```

File: src/torchmetrics/detection/diou.py [hunk 6]
```diff
@@ -110,11 +110,6 @@ def __init__(
         respect_labels: bool = True,
         **kwargs: Any,
     ) -> None:
```
Comment (Borda) [inline src/torchmetrics/detection/diou.py:112]:
```suggestion
    ) -> None:
    if not _TORCHVISION_AVAILABLE:
            raise ModuleNotFoundError(
                f"Metric `{self._iou_type.upper()}` requires that `torchvision` to be installed."
                " Please install with `pip install torchvision -U` or `pip install torchmetrics[detection]`."
            )
```

File: src/torchmetrics/detection/mean_ap.py [hunk 7]
```diff
@@ -383,11 +382,6 @@ def __init__(
                 " Please install with `pip install pycocotools` or `pip install faster-coco-eval` or"
                 " `pip install torchmetrics[detection]`."
             )
```
Comment (Borda) [inline src/torchmetrics/detection/mean_ap.py:384]:
```suggestion
            )
    if not _TORCHVISION_AVAILABLE:
            raise ModuleNotFoundError(
                f"Metric `{self._iou_type.upper()}` requires that `torchvision` to be installed."
                " Please install with `pip install torchmetrics[detection]`."
            )
```

File: src/torchmetrics/functional/detection/ciou.py [hunk 8]
```diff
@@ -113,11 +113,5 @@ def complete_intersection_over_union(
                 [-0.3971, -0.1543,  0.5606]])
 
     """
```
Comment (Borda) [inline src/torchmetrics/functional/detection/ciou.py:115]:
```suggestion
    """
    if not _TORCHVISION_AVAILABLE:
            raise ModuleNotFoundError(
                f"`{complete_intersection_over_union.__name__}` requires that `torchvision` is be installed."
                " Please install with `pip install torchmetrics[detection]`."
            )
```

File: src/torchmetrics/detection/_mean_ap.py [hunk 9]
```diff
@@ -327,11 +327,6 @@ def __init__(
                 "`MAP` metric requires that `pycocotools` installed."
                 " Please install with `pip install pycocotools` or `pip install torchmetrics[detection]`"
             )
```
Comment (Borda) [inline src/torchmetrics/detection/_mean_ap.py:329]:
```suggestion
            )
        if not _TORCHVISION_AVAILABLE:
            raise ModuleNotFoundError(
                "`MeanAveragePrecision` metric requires that `torchvision` is installed."
                " Please install with `pip install torchmetrics[detection]`."
            )
```

File: src/torchmetrics/functional/detection/diou.py [hunk 10]
```diff
@@ -113,11 +113,5 @@ def distance_intersection_over_union(
                 [-0.3971, -0.1510,  0.5609]])
 
     """
```
Comment (Borda) [inline src/torchmetrics/functional/detection/diou.py:115]:
```suggestion
    """
    if not _TORCHVISION_AVAILABLE:
            raise ModuleNotFoundError(
                f"`{complete_intersection_over_union.__name__}` requires that `torchvision` is installed."
                " Please install with `pip install torchmetrics[detection]`."
            )
```

File: src/torchmetrics/functional/detection/giou.py [hunk 11]
```diff
@@ -113,11 +113,5 @@ def generalized_intersection_over_union(
                 [-0.6024, -0.4021,  0.5345]])
 
     """
```
Comment (Borda) [inline src/torchmetrics/functional/detection/giou.py:115]:
```suggestion
    """
    if not _TORCHVISION_AVAILABLE:
            raise ModuleNotFoundError(
                f"`{complete_intersection_over_union.__name__}` requires that `torchvision` is installed."
                " Please install with `pip install torchmetrics[detection]`."
            )
```

File: src/torchmetrics/functional/detection/iou.py [hunk 12]
```diff
@@ -114,10 +114,5 @@ def intersection_over_union(
                 [0.0000, 0.0000, 0.5654]])
 
     """
```
Comment (Borda) [inline src/torchmetrics/functional/detection/iou.py:116]:
```suggestion
    """
    if not _TORCHVISION_AVAILABLE:
            raise ModuleNotFoundError(
                f"`{complete_intersection_over_union.__name__}` requires that `torchvision` is installed."
                " Please install with `pip install torchmetrics[detection]`."
            )
```

File: src/torchmetrics/detection/_mean_ap.py [hunk 13]
```diff
@@ -22,13 +22,13 @@
...
 from torchmetrics.utilities.data import _cumsum
-from torchmetrics.utilities.imports import _MATPLOTLIB_AVAILABLE, _PYCOCOTOOLS_AVAILABLE, _TORCHVISION_GREATER_EQUAL_0_8
+from torchmetrics.utilities.imports import _MATPLOTLIB_AVAILABLE, _PYCOCOTOOLS_AVAILABLE
 from torchmetrics.utilities.plot import _AX_TYPE, _PLOT_OUT_TYPE
 
 if not _MATPLOTLIB_AVAILABLE:
     __doctest_skip__ = ["MeanAveragePrecision.plot"]
 
-if not _TORCHVISION_GREATER_EQUAL_0_8 or not _PYCOCOTOOLS_AVAILABLE:
+if not _PYCOCOTOOLS_AVAILABLE:
```
Comment (Borda) [inline src/torchmetrics/detection/_mean_ap.py:31]:
```suggestion
if not _TORCHVISION_AVAILABLE or not _PYCOCOTOOLS_AVAILABLE:
```

File: src/torchmetrics/detection/_mean_ap.py [hunk 14]
```diff
@@ -22,13 +22,13 @@
 from torchmetrics.detection.helpers import _fix_empty_tensors, _input_validator
 from torchmetrics.metric import Metric
 from torchmetrics.utilities.data import _cumsum
-from torchmetrics.utilities.imports import _MATPLOTLIB_AVAILABLE, _PYCOCOTOOLS_AVAILABLE, _TORCHVISION_GREATER_EQUAL_0_8
+from torchmetrics.utilities.imports import _MATPLOTLIB_AVAILABLE, _PYCOCOTOOLS_AVAILABLE
```
Comment (Borda) [inline src/torchmetrics/detection/_mean_ap.py:25]:
```suggestion
from torchmetrics.utilities.imports import _MATPLOTLIB_AVAILABLE, _TORCHVISION_AVAILABLE, _PYCOCOTOOLS_AVAILABLE
```

File: src/torchmetrics/detection/mean_ap.py [hunk 15]
```diff
@@ -383,11 +382,11 @@ def __init__(
...
-        if not _TORCHVISION_GREATER_EQUAL_0_8:
-            raise ModuleNotFoundError(
-                "`MeanAveragePrecision` metric requires that `torchvision` version 0.8.0 or newer is installed."
-                " Please install with `pip install torchvision>=0.8` or `pip install torchmetrics[detection]`."
-            )
+	    if not _TORCHVISION_AVAILABLE:
+	            raise ModuleNotFoundError(
+	                f"Metric `{self._iou_type.upper()}` requires that `torchvision` is installed."
+	                " Please install with `pip install torchmetrics[detection]`."
+	            )
```
Comment (Borda) [inline src/torchmetrics/detection/mean_ap.py:389]:
```suggestion
	if not _TORCHVISION_AVAILABLE:
	        raise ModuleNotFoundError(
	            f"Metric `{self._iou_type.upper()}` requires that `torchvision` is installed."
	            " Please install with `pip install torchmetrics[detection]`."
	        )
```

File: src/torchmetrics/functional/image/lpips.py [hunk 16]
```diff
@@ -52,13 +52,10 @@ def _get_net(net: str, pretrained: bool) -> nn.modules.container.Sequential:
     """
     from torchvision import models as tv
 
-    if _TORCHVISION_GREATER_EQUAL_0_13:
-        if pretrained:
-            pretrained_features = getattr(tv, net)(weights=getattr(tv, _weight_map[net]).IMAGENET1K_V1).features
-        else:
-            pretrained_features = getattr(tv, net)(weights=None).features
+    if pretrained:
+        pretrained_features = getattr(tv, net)(weights=getattr(tv, _weight_map[net]).IMAGENET1K_V1).features
```
Comment (Borda) [inline src/torchmetrics/functional/image/lpips.py:56]:
```suggestion
    if _TORCHVISION_AVAILABLE
        if pretrained:
            pretrained_features = getattr(tv, net)(weights=getattr(tv, _weight_map[net]).IMAGENET1K_V1).features
```

File: src/torchmetrics/functional/image/lpips.py [hunk 17]
```diff
@@ -52,13 +52,10 @@ def _get_net(net: str, pretrained: bool) -> nn.modules.container.Sequential:
...
     else:
-        pretrained_features = getattr(tv, net)(pretrained=pretrained).features
+        pretrained_features = getattr(tv, net)(weights=None).features
```
Comment (Borda) [inline src/torchmetrics/functional/image/lpips.py:58]:
```suggestion
            pretrained_features = getattr(tv, net)(weights=None).features
```

File: src/torchmetrics/functional/image/lpips.py [hunk 18]
```diff
@@ -52,13 +52,10 @@ def _get_net(net: str, pretrained: bool) -> nn.modules.container.Sequential:
...
     from torchvision import models as tv
 
-    if _TORCHVISION_GREATER_EQUAL_0_13:
-        if pretrained:
-            pretrained_features = getattr(tv, net)(weights=getattr(tv, _weight_map[net]).IMAGENET1K_V1).features
-        else:
-            pretrained_features = getattr(tv, net)(weights=None).features
+    if pretrained:
+        pretrained_features = getattr(tv, net)(weights=getattr(tv, _weight_map[net]).IMAGENET1K_V1).features
     else:
```
Comment (Borda) [inline src/torchmetrics/functional/image/lpips.py:57]:
```suggestion
        else:
```

**Comment by SkafteNicki** [comment]
@Borda what is missing here?

**Comment by Borda** [comment]
> what is missing here?

just last check and also we still hit this error:
```
Traceback (most recent call last):
  File "/__w/2/s/tests/unittests/nominal/test_cramers.py", line 64, in _reference_dython_cramers_v
    from dython.nominal import cramers_v
  File "/usr/local/lib/python3.10/dist-packages/dython/__init__.py", line 1, in <module>
    from . import nominal, model_utils, sampling, data_utils
  File "/usr/local/lib/python3.10/dist-packages/dython/model_utils.py", line 4, in <module>
    from scikitplot.helpers import binary_ks_curve
  File "/usr/local/lib/python3.10/dist-packages/scikitplot/__init__.py", line 2, in <module>
    from . import metrics, cluster, decomposition, estimators
  File "/usr/local/lib/python3.10/dist-packages/scikitplot/metrics.py", line 27, in <module>
    from scipy import interp
ImportError: cannot import name 'interp' from 'scipy' (/usr/local/lib/python3.10/dist-packages/scipy/__init__.py)
```

**Comment by SkafteNicki** [comment]
@Borda install `scipy==1.11.4` according to this issue: https://github.com/reiinakano/scikit-plot/issues/119

**Comment by SkafteNicki** [comment]
Alternative set `dython>=0.7.5` as according to this issue: https://github.com/shakedzy/dython/issues/156

**Comment by Borda** [comment]
> @Borda install `scipy==1.11.4` according to this issue: [reiinakano/scikit-plot#119](https://github.com/reiinakano/scikit-plot/issues/119)

wanted to avoid this freeze in tests...

> Alternative set `dython>=0.7.5` as according to this issue: [shakedzy/dython#156](https://github.com/shakedzy/dython/issues/156)

I did free to latest `0.7.7` and bump `matplotlib >=3.6.0` which is still 2022

**Comment by SkafteNicki** [comment]
@Borda yeah I also think it is time to revisit this. Will help also land another PR https://github.com/Lightning-AI/torchmetrics/pull/2777#issuecomment-2415024910 which is currently blocked by old PT versions

**Review by lantiga — APPROVED** [review]

**Pull request merged**